src/mixins/datepicker-calendar-mixin/datepicker-calendar-mixin.ts

```diff
@@ -0,0 +1,114 @@
+import { LitElement } from "lit";
+import { property } from "lit/decorators.js";
+import { CALENDAR_TYPES } from "../../components/calendar/bl-calendar.constant";
+import { CalendarType, DayValues } from "../../components/calendar/bl-calendar.types";
+
+export default class DatepickerCalendarMixin extends LitElement {
+  /**
+   * Defines the calendar types, available types are single, multiple and range
+   */
+  @property()
+  type: CalendarType;
+  /**
+   * Defines the start day of the calendar (1 defines monday)
+   */
+  @property({ type: Number, attribute: "start-of-week", reflect: true })
+  startOfWeek: DayValues = 0;
+  /**
+   * Defines the calendar language
+   */
+  @property()
+  locale: string = document.documentElement.lang || "en-EN";
+
+  /**
+   * Defines the unselectable dates for calendar
+   */
+  _disabledDates: Date[] = [];
+  get disabledDates(): Date[] {
+    return this._disabledDates;
+  }
+
+  @property({ attribute: "disabled-dates", reflect: true })
+  set disabledDates(disabledDates: Date[] | string) {
+    if (typeof disabledDates === "string") {
+      const splitDisabledDates = disabledDates.split(",");
+
+      splitDisabledDates?.forEach(disabledDate => {
+        const date = new Date(`${disabledDate}T00:00:00`);
+
+        if (!isNaN(date.getTime())) this._disabledDates.push(date);
+      });
+    } else if (Array.isArray(disabledDates)) {
+      disabledDates.forEach(disabledDate => {
+        this._disabledDates.push(disabledDate);
+      });
+    } else {
+      console.warn("invalid disabledDate format.DisabledDates should be string or Date array.");
+    }
```

Comment:
We can extract this into a function and use it in all fields that require a common result.